### ISSUE TITLE
GPG Agent Hangs on init command and appears to be poisoned whne trying to use gpg

### ISSUE THREAD

**Issue by sittingfrog** [romanz/trezor-agent#351]
trezor-agent is instlal on macos and  the ssh agent works but gpg does not.

When attempting to initialize, the initialization works but hangs at the end and has to be manually escaped.

When attempting to sign a text file or add a key or do anything else with gpg, there is a stale lock file error which may be related to poisoned keys. However, I don't fully understand this because the key should theoretically be brand new (although i noticed that the keygen process defaults to `--time 0`). Then again, I'm very new to gpg.

```2021-05-02 18:19:54,480 DEBUG        sys.argv: ['/usr/local/bin/trezor-gpg-agent', '-vv', '--pin-entry-binary=pinentry', '--passphrase-entry-binary=pinentry', '--cache-expiry-seconds=inf', '--homedir', '/Users/jn/.gnupg/trezor', '--use-standard-socket', '--daemon'] [__init__.py:244]
2021-05-02 18:19:54,480 DEBUG        os.environ: environ({'export': '', 'TERM_PROGRAM': 'iTerm.app', 'TERM': 'xterm-256color', 'SHELL': '/bin/zsh', 'APPLE_ID': 'jvj.norton@gmail.com', 'TMPDIR': '/var/folders/hb/ygcz9qz97x502lx3kl2zzs940000gn/T/', 'DOCKER_HOST': 'tcp://192.168.99.100:2376', 'TERM_PROGRAM_VERSION': '3.4.6', 'APPLE_LAST_NAME': 'Norton', 'TERM_SESSION_ID': 'w0t0p0:080D616A-59E2-4A72-9982-BD5C1E72997E', 'ZSH': '/Users/jn/.oh-my-zsh', 'USER': 'jn', 'COMMAND_MODE': 'unix2003', 'DOCKER_MACHINE_NAME': 'default', 'SSH_AUTH_SOCK': '/private/tmp/com.apple.launchd.NS2YVwacwX/Listeners', '__CF_USER_TEXT_ENCODING': '0x1F5:0x0:0x52', 'PAGER': 'less', 'DOCKER_TLS_VERIFY': '1', 'LOCAL_SSH_AUTH_SOCK': '/private/tmp/com.apple.launchd.NS2YVwacwX/Listeners', 'LSCOLORS': 'Gxfxcxdxbxegedabagacad', 'PATH': '/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin', 'LaunchInstanceID': '9E0DB4A8-DF7A-4A7F-8CA8-B66F5DA763D0', '__CFBundleIdentifier': 'com.googlecode.iterm2', 'PWD': '/Users/jn', 'LANG': 'en_CA.UTF-8', 'ITERM_PROFILE': 'Default', 'XPC_FLAGS': '0x0', 'XPC_SERVICE_NAME': '0', 'GPG_TTY': '/dev/ttys000', 'DOCKER_CERT_PATH': '/Users/jn/.docker/machine/machines/default', 'SHLVL': '2', 'HOME': '/Users/jn', 'COLORFGBG': '7;0', 'TREZORCTL': '/Users/jn/code/trezor-firmware/.venv/bin/trezorctl', 'LC_TERMINAL_VERSION': '3.4.6', 'ITERM_SESSION_ID': 'w0t0p0:080D616A-59E2-4A72-9982-BD5C1E72997E', 'LESS': '-R', 'APPLE_FIRST_NAME': 'Jonathan', 'LOGNAME': 'jn', 'LC_TERMINAL': 'iTerm2', 'SECURITYSESSIONID': '186a9', 'GNUPGHOME': '/Users/jn/.gnupg/trezor', 'COLORTERM': 'truecolor', '_': '/usr/local/bin/trezor-gpg-agent'}) [__init__.py:245]
2021-05-02 18:19:54,481 DEBUG        pid: 19879, parent pid: 19878                                                                        [__init__.py:246]
2021-05-02 18:19:54,481 DEBUG        which 'gpgconf' => '/usr/local/bin/gpgconf'                                                          [util.py:249]
2021-05-02 18:19:54,481 DEBUG        calling ['/usr/local/bin/gpgconf', '--list-components'] with env None                                [keyring.py:19]
2021-05-02 18:19:54,485 DEBUG        output: b'gpg:OpenPGP:/usr/local/Cellar/gnupg/2.3.1/bin/gpg\ngpgsm:S/MIME:/usr/local/Cellar/gnupg/2.3.1/bin/gpgsm\nkeyboxd:Public Keys:/usr/local/Cellar/gnupg/2.3.1/libexec/keyboxd\ngpg-agent:Private Keys:/usr/local/Cellar/gnupg/2.3.1/bin/gpg-agent\nscdaemon:Smartcards:/usr/local/Cellar/gnupg/2.3.1/libexec/scdaemon\ntpm2daemon:TPM:/usr/local/Cellar/gnupg/2.3.1/libexec/tpm2daemon\ndirmngr:Network:/usr/local/Cellar/gnupg/2.3.1/bin/dirmngr\npinentry:Passphrase Entry:/usr/local/opt/pinentry/bin/pinentry\n' [keyring.py:21]
2021-05-02 18:19:54,486 DEBUG        gpgconf --list-components: {'gpg': '/usr/local/Cellar/gnupg/2.3.1/bin/gpg', 'gpgsm': '/usr/local/Cellar/gnupg/2.3.1/bin/gpgsm', 'keyboxd': '/usr/local/Cellar/gnupg/2.3.1/libexec/keyboxd', 'gpg-agent': '/usr/local/Cellar/gnupg/2.3.1/bin/gpg-agent', 'scdaemon': '/usr/local/Cellar/gnupg/2.3.1/libexec/scdaemon', 'tpm2daemon': '/usr/local/Cellar/gnupg/2.3.1/libexec/tpm2daemon', 'dirmngr': '/usr/local/Cellar/gnupg/2.3.1/bin/dirmngr', 'pinentry': '/usr/local/opt/pinentry/bin/pinentry'} [keyring.py:194]
2021-05-02 18:19:54,486 DEBUG        calling ['/usr/local/Cellar/gnupg/2.3.1/bin/gpg', '--export'] with env {'GNUPGHOME': '/Users/jn/.gnupg/trezor', 'PATH': '/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin'} [keyring.py:19]
2021-05-02 18:19:54,496 DEBUG        output: b'\x98R\x04\x00\x00\x00\x00\x13\x08*\x86H\xce=\x03\x01\x07\x02\x03\x04,e\xe3&\x0cu\xe9(\xd9\xf4\xdc\x96a\x97(\xb8\xab\xfb\xa58q\xce\x9b\xbc_\xfe\x0e\xe3\xb1c\xb9\xe7\xd4\xcb\xafW\xf82|\x1b\xac\xae=e$k\xea\xf3Q\xd3\xbdS\xfeC]\x95%\x087X\xe6n;\x83\xb4&Jonathan Norton <jvj.norton@gmail.com>\x88\x7f\x04\x13\x13\x08\x00\x1c\x05\x02\x00\x00\x00\x00\x02\x0b\t\x02\x1b\x03\x04\x15\x08\t\n\x04\x16\x02\x03\x01\x02\x17\x80\x02\x1e\x01\x00\x16\t\x10Q\xf3\x02\xfcJ\x80\xc3\xaf\x0b\x1aTREZOR-GPG\x87\x8d\x00\xf5\x122\xf1\xaf\x0c\x05,Q\xd2*\xeb\x90\x8e9\xa3\xc4\xd5Q\x93*A\xa3\xc8\x13W\xd3\x84\xc2AL\xb2\x00\xfff\x90x`;\x10\xfb\xa05\x12xK\xd8b4j\x17\xad\x1a\x94\xf05\x86\x13\x12]qVc\xc92\x9d\xb8V\x04\x00\x00\x00\x00\x12\x08*\x86H\xce=\x03\x01\x07\x02\x03\x04\xfbNv\x89m\xf4\x0f\xf1\x0eo\x15G*f\xe9\xa0\x85\x03\x90\xf0\x19\xe6p\xc9\x12E\xb7,\x11B\x98\xe8nA\xe9\x18\x03|\x00\x9a\xa4\xc0w\xeb\xc2\xb7&\xe0=\xecLt\x01\xc2h\xdeK\xa2\xdeG\x83n\xc1V\x03\x01\x08\x07\x88m\x04\x18\x13\x08\x00\t\x05\x02\x00\x00\x00\x00\x02\x1b\x0c\x00\x16\t\x10Q\xf3\x02\xfcJ\x80\xc3\xaf\x0b\x1aTREZOR-GPG\xee\xfd\x00\xffi-\x173\xf1\xba\xc3.\x03\xcb\xc4Hw\xc8U\xf6\xc9\xf0\x07\x08\xd7\xaa4W\xb0\x90\nS\x85$\xda\xb1\x00\xffu\xbf>\x80VRF\x90u\x98\xf4~\xf9J\xcc\\k\x1c\xcc\xf2\xae\xb6\xc1\xe2)\xce\x8d|\xd3d\x15\xda' [keyring.py:21]
2021-05-02 18:19:54,499 WARNING      no TTY found: Command '['tty']' returned non-zero exit status 1.                                     [ui.py:83]
2021-05-02 18:19:54,500 WARNING      DISPLAY not defined                                                                                  [ui.py:89]
2021-05-02 18:19:54,500 INFO         using [] for pinentry options                                                                        [ui.py:91]
2021-05-02 18:19:54,500 DEBUG        calling ['/usr/local/Cellar/gnupg/2.3.1/bin/gpg', '--version'] with env None                         [keyring.py:19]
2021-05-02 18:19:54,505 DEBUG        output: b'gpg (GnuPG) 2.3.1\nlibgcrypt 1.9.3\nCopyright (C) 2021 Free Software Foundation, Inc.\nLicense GNU GPL-3.0-or-later <https://gnu.org/licenses/gpl.html>\nThis is free software: you are free to change and redistribute it.\nThere is NO WARRANTY, to the extent permitted by law.\n\nHome: /Users/jn/.gnupg/trezor\nSupported algorithms:\nPubkey: RSA, ELG, DSA, ECDH, ECDSA, EDDSA\nCipher: IDEA, 3DES, CAST5, BLOWFISH, AES, AES192, AES256, TWOFISH,\n        CAMELLIA128, CAMELLIA192, CAMELLIA256\nAEAD: EAX, OCB\nHash: SHA1, RIPEMD160, SHA256, SHA384, SHA512, SHA224\nCompression: Uncompressed, ZIP, ZLIB, BZIP2\n' [keyring.py:21]
2021-05-02 18:19:54,506 DEBUG        calling ['/usr/local/bin/gpgconf', '--list-dirs'] with env {'GNUPGHOME': '/Users/jn/.gnupg/trezor', 'PATH': '/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin'} [keyring.py:19]
2021-05-02 18:19:54,514 DEBUG        output: b'sysconfdir:/usr/local/etc/gnupg\nbindir:/usr/local/Cellar/gnupg/2.3.1/bin\nlibexecdir:/usr/local/Cellar/gnupg/2.3.1/libexec\nlibdir:/usr/local/Cellar/gnupg/2.3.1/lib/gnupg\ndatadir:/usr/local/Cellar/gnupg/2.3.1/share/gnupg\nlocaledir:/usr/local/Cellar/gnupg/2.3.1/share/locale\nsocketdir:/Users/jn/.gnupg/trezor\ndirmngr-socket:/Users/jn/.gnupg/trezor/S.dirmngr\nkeyboxd-socket:/Users/jn/.gnupg/trezor/S.keyboxd\nagent-ssh-socket:/Users/jn/.gnupg/trezor/S.gpg-agent.ssh\nagent-extra-socket:/Users/jn/.gnupg/trezor/S.gpg-agent.extra\nagent-browser-socket:/Users/jn/.gnupg/trezor/S.gpg-agent.browser\nagent-socket:/Users/jn/.gnupg/trezor/S.gpg-agent\nhomedir:/Users/jn/.gnupg/trezor\n' [keyring.py:21]
2021-05-02 18:19:54,514 DEBUG        ['/usr/local/bin/gpgconf', '--list-dirs']: {b'sysconfdir': b'/usr/local/etc/gnupg', b'bindir': b'/usr/local/Cellar/gnupg/2.3.1/bin', b'libexecdir': b'/usr/local/Cellar/gnupg/2.3.1/libexec', b'libdir': b'/usr/local/Cellar/gnupg/2.3.1/lib/gnupg', b'datadir': b'/usr/local/Cellar/gnupg/2.3.1/share/gnupg', b'localedir': b'/usr/local/Cellar/gnupg/2.3.1/share/locale', b'socketdir': b'/Users/jn/.gnupg/trezor', b'dirmngr-socket': b'/Users/jn/.gnupg/trezor/S.dirmngr', b'keyboxd-socket': b'/Users/jn/.gnupg/trezor/S.keyboxd', b'agent-ssh-socket': b'/Users/jn/.gnupg/trezor/S.gpg-agent.ssh', b'agent-extra-socket': b'/Users/jn/.gnupg/trezor/S.gpg-agent.extra', b'agent-browser-socket': b'/Users/jn/.gnupg/trezor/S.gpg-agent.browser', b'agent-socket': b'/Users/jn/.gnupg/trezor/S.gpg-agent', b'homedir': b'/Users/jn/.gnupg/trezor'} [keyring.py:31]
2021-05-02 18:19:54,515 DEBUG        serving on b'/Users/jn/.gnupg/trezor/S.gpg-agent'                                                    [server.py:30]
2021-05-02 18:19:54,515 DEBUG        waiting for connection on /Users/jn/.gnupg/trezor/S.gpg-agent                                        [agent.py:14]
2021-05-02 18:22:56,328 DEBUG        sys.argv: ['/usr/local/bin/trezor-gpg-agent', '-vv', '--pin-entry-binary=pinentry', '--passphrase-entry-binary=pinentry', '--cache-expiry-seconds=inf', '--homedir', '/Users/jn/.gnupg/trezor', '--use-standard-socket', '--daemon'] [__init__.py:244]
2021-05-02 18:22:56,328 DEBUG        os.environ: environ({'export': '', 'TERM_PROGRAM': 'iTerm.app', 'TERM': 'xterm-256color', 'SHELL': '/bin/zsh', 'APPLE_ID': 'jvj.norton@gmail.com', 'TMPDIR': '/var/folders/hb/ygcz9qz97x502lx3kl2zzs940000gn/T/', 'DOCKER_HOST': 'tcp://192.168.99.100:2376', 'TERM_PROGRAM_VERSION': '3.4.6', 'APPLE_LAST_NAME': 'Norton', 'TERM_SESSION_ID': 'w0t0p0:080D616A-59E2-4A72-9982-BD5C1E72997E', 'ZSH': '/Users/jn/.oh-my-zsh', 'USER': 'jn', 'COMMAND_MODE': 'unix2003', 'DOCKER_MACHINE_NAME': 'default', 'SSH_AUTH_SOCK': '/private/tmp/com.apple.launchd.NS2YVwacwX/Listeners', '__CF_USER_TEXT_ENCODING': '0x1F5:0x0:0x52', 'PAGER': 'less', 'DOCKER_TLS_VERIFY': '1', 'LOCAL_SSH_AUTH_SOCK': '/private/tmp/com.apple.launchd.NS2YVwacwX/Listeners', 'LSCOLORS': 'Gxfxcxdxbxegedabagacad', 'PATH': '/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin', 'LaunchInstanceID': '9E0DB4A8-DF7A-4A7F-8CA8-B66F5DA763D0', '__CFBundleIdentifier': 'com.googlecode.iterm2', 'PWD': '/Users/jn', 'LANG': 'en_CA.UTF-8', 'ITERM_PROFILE': 'Default', 'XPC_FLAGS': '0x0', 'XPC_SERVICE_NAME': '0', 'GPG_TTY': '/dev/ttys000', 'DOCKER_CERT_PATH': '/Users/jn/.docker/machine/machines/default', 'SHLVL': '2', 'HOME': '/Users/jn', 'COLORFGBG': '7;0', 'TREZORCTL': '/Users/jn/code/trezor-firmware/.venv/bin/trezorctl', 'LC_TERMINAL_VERSION': '3.4.6', 'ITERM_SESSION_ID': 'w0t0p0:080D616A-59E2-4A72-9982-BD5C1E72997E', 'LESS': '-R', 'APPLE_FIRST_NAME': 'Jonathan', 'LOGNAME': 'jn', 'LC_TERMINAL': 'iTerm2', 'SECURITYSESSIONID': '186a9', 'GNUPGHOME': '/Users/jn/.gnupg/trezor', 'COLORTERM': 'truecolor', '_': '/usr/local/bin/trezor-gpg-agent'}) [__init__.py:245]
2021-05-02 18:22:56,329 DEBUG        pid: 19901, parent pid: 19900                                                                        [__init__.py:246]
2021-05-02 18:22:56,329 DEBUG        which 'gpgconf' => '/usr/local/bin/gpgconf'                                                          [util.py:249]
2021-05-02 18:22:56,329 DEBUG        calling ['/usr/local/bin/gpgconf', '--list-components'] with env None                                [keyring.py:19]
2021-05-02 18:22:56,333 DEBUG        output: b'gpg:OpenPGP:/usr/local/Cellar/gnupg/2.3.1/bin/gpg\ngpgsm:S/MIME:/usr/local/Cellar/gnupg/2.3.1/bin/gpgsm\nkeyboxd:Public Keys:/usr/local/Cellar/gnupg/2.3.1/libexec/keyboxd\ngpg-agent:Private Keys:/usr/local/Cellar/gnupg/2.3.1/bin/gpg-agent\nscdaemon:Smartcards:/usr/local/Cellar/gnupg/2.3.1/libexec/scdaemon\ntpm2daemon:TPM:/usr/local/Cellar/gnupg/2.3.1/libexec/tpm2daemon\ndirmngr:Network:/usr/local/Cellar/gnupg/2.3.1/bin/dirmngr\npinentry:Passphrase Entry:/usr/local/opt/pinentry/bin/pinentry\n' [keyring.py:21]
2021-05-02 18:22:56,334 DEBUG        gpgconf --list-components: {'gpg': '/usr/local/Cellar/gnupg/2.3.1/bin/gpg', 'gpgsm': '/usr/local/Cellar/gnupg/2.3.1/bin/gpgsm', 'keyboxd': '/usr/local/Cellar/gnupg/2.3.1/libexec/keyboxd', 'gpg-agent': '/usr/local/Cellar/gnupg/2.3.1/bin/gpg-agent', 'scdaemon': '/usr/local/Cellar/gnupg/2.3.1/libexec/scdaemon', 'tpm2daemon': '/usr/local/Cellar/gnupg/2.3.1/libexec/tpm2daemon', 'dirmngr': '/usr/local/Cellar/gnupg/2.3.1/bin/dirmngr', 'pinentry': '/usr/local/opt/pinentry/bin/pinentry'} [keyring.py:194]
2021-05-02 18:22:56,334 DEBUG        calling ['/usr/local/Cellar/gnupg/2.3.1/bin/gpg', '--export'] with env {'GNUPGHOME': '/Users/jn/.gnupg/trezor', 'PATH': '/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin'} [keyring.py:19]
2021-05-02 18:22:56,344 DEBUG        output: b'\x98R\x04\x00\x00\x00\x00\x13\x08*\x86H\xce=\x03\x01\x07\x02\x03\x04,e\xe3&\x0cu\xe9(\xd9\xf4\xdc\x96a\x97(\xb8\xab\xfb\xa58q\xce\x9b\xbc_\xfe\x0e\xe3\xb1c\xb9\xe7\xd4\xcb\xafW\xf82|\x1b\xac\xae=e$k\xea\xf3Q\xd3\xbdS\xfeC]\x95%\x087X\xe6n;\x83\xb4&Jonathan Norton <jvj.norton@gmail.com>\x88\x7f\x04\x13\x13\x08\x00\x1c\x05\x02\x00\x00\x00\x00\x02\x0b\t\x02\x1b\x03\x04\x15\x08\t\n\x04\x16\x02\x03\x01\x02\x17\x80\x02\x1e\x01\x00\x16\t\x10Q\xf3\x02\xfcJ\x80\xc3\xaf\x0b\x1aTREZOR-GPG\x87\x8d\x00\xf5\x122\xf1\xaf\x0c\x05,Q\xd2*\xeb\x90\x8e9\xa3\xc4\xd5Q\x93*A\xa3\xc8\x13W\xd3\x84\xc2AL\xb2\x00\xfff\x90x`;\x10\xfb\xa05\x12xK\xd8b4j\x17\xad\x1a\x94\xf05\x86\x13\x12]qVc\xc92\x9d\xb8V\x04\x00\x00\x00\x00\x12\x08*\x86H\xce=\x03\x01\x07\x02\x03\x04\xfbNv\x89m\xf4\x0f\xf1\x0eo\x15G*f\xe9\xa0\x85\x03\x90\xf0\x19\xe6p\xc9\x12E\xb7,\x11B\x98\xe8nA\xe9\x18\x03|\x00\x9a\xa4\xc0w\xeb\xc2\xb7&\xe0=\xecLt\x01\xc2h\xdeK\xa2\xdeG\x83n\xc1V\x03\x01\x08\x07\x88m\x04\x18\x13\x08\x00\t\x05\x02\x00\x00\x00\x00\x02\x1b\x0c\x00\x16\t\x10Q\xf3\x02\xfcJ\x80\xc3\xaf\x0b\x1aTREZOR-GPG\xee\xfd\x00\xffi-\x173\xf1\xba\xc3.\x03\xcb\xc4Hw\xc8U\xf6\xc9\xf0\x07\x08\xd7\xaa4W\xb0\x90\nS\x85$\xda\xb1\x00\xffu\xbf>\x80VRF\x90u\x98\xf4~\xf9J\xcc\\k\x1c\xcc\xf2\xae\xb6\xc1\xe2)\xce\x8d|\xd3d\x15\xda' [keyring.py:21]
2021-05-02 18:22:56,347 WARNING      no TTY found: Command '['tty']' returned non-zero exit status 1.                                     [ui.py:83]
2021-05-02 18:22:56,347 WARNING      DISPLAY not defined                                                                                  [ui.py:89]
2021-05-02 18:22:56,347 INFO         using [] for pinentry options                                                                        [ui.py:91]
2021-05-02 18:22:56,347 DEBUG        calling ['/usr/local/Cellar/gnupg/2.3.1/bin/gpg', '--version'] with env None                         [keyring.py:19]
2021-05-02 18:22:56,353 DEBUG        output: b'gpg (GnuPG) 2.3.1\nlibgcrypt 1.9.3\nCopyright (C) 2021 Free Software Foundation, Inc.\nLicense GNU GPL-3.0-or-later <https://gnu.org/licenses/gpl.html>\nThis is free software: you are free to change and redistribute it.\nThere is NO WARRANTY, to the extent permitted by law.\n\nHome: /Users/jn/.gnupg/trezor\nSupported algorithms:\nPubkey: RSA, ELG, DSA, ECDH, ECDSA, EDDSA\nCipher: IDEA, 3DES, CAST5, BLOWFISH, AES, AES192, AES256, TWOFISH,\n        CAMELLIA128, CAMELLIA192, CAMELLIA256\nAEAD: EAX, OCB\nHash: SHA1, RIPEMD160, SHA256, SHA384, SHA512, SHA224\nCompression: Uncompressed, ZIP, ZLIB, BZIP2\n' [keyring.py:21]
2021-05-02 18:22:56,353 DEBUG        calling ['/usr/local/bin/gpgconf', '--list-dirs'] with env {'GNUPGHOME': '/Users/jn/.gnupg/trezor', 'PATH': '/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin'} [keyring.py:19]
2021-05-02 18:22:56,362 DEBUG        output: b'sysconfdir:/usr/local/etc/gnupg\nbindir:/usr/local/Cellar/gnupg/2.3.1/bin\nlibexecdir:/usr/local/Cellar/gnupg/2.3.1/libexec\nlibdir:/usr/local/Cellar/gnupg/2.3.1/lib/gnupg\ndatadir:/usr/local/Cellar/gnupg/2.3.1/share/gnupg\nlocaledir:/usr/local/Cellar/gnupg/2.3.1/share/locale\nsocketdir:/Users/jn/.gnupg/trezor\ndirmngr-socket:/Users/jn/.gnupg/trezor/S.dirmngr\nkeyboxd-socket:/Users/jn/.gnupg/trezor/S.keyboxd\nagent-ssh-socket:/Users/jn/.gnupg/trezor/S.gpg-agent.ssh\nagent-extra-socket:/Users/jn/.gnupg/trezor/S.gpg-agent.extra\nagent-browser-socket:/Users/jn/.gnupg/trezor/S.gpg-agent.browser\nagent-socket:/Users/jn/.gnupg/trezor/S.gpg-agent\nhomedir:/Users/jn/.gnupg/trezor\n' [keyring.py:21]
2021-05-02 18:22:56,363 DEBUG        ['/usr/local/bin/gpgconf', '--list-dirs']: {b'sysconfdir': b'/usr/local/etc/gnupg', b'bindir': b'/usr/local/Cellar/gnupg/2.3.1/bin', b'libexecdir': b'/usr/local/Cellar/gnupg/2.3.1/libexec', b'libdir': b'/usr/local/Cellar/gnupg/2.3.1/lib/gnupg', b'datadir': b'/usr/local/Cellar/gnupg/2.3.1/share/gnupg', b'localedir': b'/usr/local/Cellar/gnupg/2.3.1/share/locale', b'socketdir': b'/Users/jn/.gnupg/trezor', b'dirmngr-socket': b'/Users/jn/.gnupg/trezor/S.dirmngr', b'keyboxd-socket': b'/Users/jn/.gnupg/trezor/S.keyboxd', b'agent-ssh-socket': b'/Users/jn/.gnupg/trezor/S.gpg-agent.ssh', b'agent-extra-socket': b'/Users/jn/.gnupg/trezor/S.gpg-agent.extra', b'agent-browser-socket': b'/Users/jn/.gnupg/trezor/S.gpg-agent.browser', b'agent-socket': b'/Users/jn/.gnupg/trezor/S.gpg-agent', b'homedir': b'/Users/jn/.gnupg/trezor'} [keyring.py:31]
2021-05-02 18:22:56,363 DEBUG        serving on b'/Users/jn/.gnupg/trezor/S.gpg-agent'                                                    [server.py:30]
2021-05-02 18:22:56,363 DEBUG        waiting for connection on /Users/jn/.gnupg/trezor/S.gpg-agent                                        [agent.py:14]
```

**Comment by onlykey** [comment]
@sittingfrog Are you running GnuPG 2.3.X?
gpg --version
gpg (GnuPG) 2.3.1

I think I may be seeing the same thing.

**Comment by onlykey** [comment]
@romanz Any ideas here? The issue seems to be with (GnuPG) 2.3.1 on MacOS, have not tested on Linux yet looks like to do that would need to build gpg from source.

**Comment by romanz** [comment]
I am running gpg 2.2.x on my machine - will try to build the latest version to see what the issue may be.

**Comment by romanz** [comment]
Actually, all current Ubuntu releases use gnupg 2.2.x:
https://packages.ubuntu.com/search?keywords=gnupg

**Comment by romanz** [comment]
Did you use `~/.gnupg/trezor/env`?
It should set the correct value for `GNUPGHOME`, which is used by `gpg` binary to connect to the right GPG agent:
```
#!/bin/bash
set -eu
export GNUPGHOME=/home/user/.gnupg/trezor/
COMMAND=$*
if [ -z "${COMMAND}" ]
then
    ${SHELL}
else
    ${COMMAND}
fi
```

**Comment by onlykey** [comment]
> Did you use ~/.gnupg/trezor/env?

Yes, the issue here is only with gnupg 2.3.x, as you mentioned Ubuntu releases use gnupg 2.2.x so there is no problem with the released version. I have not had a chance to build 2.3.x from source yet on Linux to see if that has the same issue as with Mac on gnupg 2.3.x.

**Comment by romanz** [comment]
Fixed in #365
